src/org/zaproxy/zap/extension/pscanrulesAlpha/UsernameIdorScanner.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import net.htmlparser.jericho.Source;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+import org.zaproxy.zap.extension.users.ExtensionUserManagement;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.users.User;
+
+public class UsernameIdorScanner extends PluginPassiveScanner {
+
+	private static final String MESSAGE_PREFIX = "pscanalpha.usernameidor.";
+	private static final int PLUGIN_ID = 10057;
+
+	private PassiveScanThread parent = null;
+	private static final Logger LOGGER = Logger.getLogger(UsernameIdorScanner.class);
+
+	private List<User> usersList = new ArrayList<User>();
+
+	private ExtensionUserManagement extUserMgmt;
+
+	@Override
+	public void setParent(PassiveScanThread parent) {
+		this.parent = parent;
+	}
+
+	private boolean shouldContinue() {
+		return !getUsers().isEmpty(); // Should continue if not empty
+	}
+
+	private List<User> getUsers() {
+//		setUsers("guest");// Uncomment for testing/debugging purposes [guest MD5: 084e0343a0486ff05530df6c705c8bb4]
+		if (usersList.isEmpty()) {

```

Comment:
It should obtain the latest users, always, otherwise changes done to the users will not be reflected in the scanner. For testing purposes it could allow to set/change `usersList` and use that if non-null, otherwise fallback to obtain the users from the contexts (returning that without changing `usersList`)?